src/components/CreateAddress/HardwareWalletPublicKeyImporter.jsx

```diff
@@ -16,34 +16,102 @@ import { Button, TextField, FormHelperText, Box, Grid } from "@mui/material";
 
 import InteractionMessages from "../InteractionMessages";
 
-class HardwareWalletPublicKeyImporter extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      bip32Path: props.defaultBIP32Path,
-      publicKeyError: "",
-      bip32PathError: "",
-      status: this.interaction().isSupported() ? PENDING : UNSUPPORTED,
-    };
-  }
-
-  interaction = () => {
-    const { network, method, defaultBIP32Path } = this.props;
-    const { bip32Path } = this.state;
+const HardwareWalletPublicKeyImporter = ({
+  enableChangeMethod,
+  disableChangeMethod,
+  validatePublicKey,
+  onImport,
+  network,
+  method,
+  defaultBIP32Path,
+}) => {
+  const [bip32Path, setBip32Path] = useState(defaultBIP32Path);
+  const [bip32PathError, setBip32PathError] = useState("");
+  const [publicKeyError, setPublicKeyError] = useState("");
+
+  const interaction = () => {
     return ExportPublicKey({
       network,
       keystore: method,
       bip32Path: bip32Path ?? defaultBIP32Path,
     });
   };
```

Comment:
Something to QA just for confirmation but I wonder if this would benefit from a `useMemo`. Basically we should only reset the interaction if network, method or bip32Path change.

Reply:
Change made in [4c7681f](https://github.com/unchained-capital/caravan/pull/342/commits/4c7681fe2badd60264d2566255b8660f7e6c4151)